ext/node/polyfills/process.ts

```diff
@@ -802,15 +802,12 @@ function processOnError(event: ErrorEvent) {
   uncaughtExceptionHandler(event.error, "uncaughtException");
 }
 
-function processOnBeforeUnload(event: Event) {
+function dispatchProcessBeforeExitEvent() {
   process.emit("beforeExit", process.exitCode || 0);
   processTicksAndRejections();
-  if (core.eventLoopHasMoreWork()) {
-    event.preventDefault();
-  }
```

Comment:
This condition seems important - I think we will need to return a boolean from this function to indicate if more work was dispatched (similar to `dispatchBeforeunloadEvent`)